### PR TITLE
features & improvements

### PR DESCRIPTION
**Pull request by abstrekt**
new PR for #1576

### CODE REVIEW

File: public/src/listeners.js [hunk 1]
```diff
@@ -394,6 +399,47 @@ export default function startListeners() {
...
+            $('#element').val('');
+            $('#menu').css('display', 'block');
+            $('#filter').css('display', 'none');
+            $('.filteX').css('display', 'none');
+        });
+        const value = $(this).val().toLowerCase();
+        if (value.length === 0) {
+            $('#filter').css('display', 'none');
+            $('.filterX').css('display', 'none');
+            $('#filter').empty();
```
Comment (revathskumar) [inline public/src/listeners.js:423]: can chain this with above `$('#filter').css('display', 'none').empty()`

Comment (revathskumar) [inline public/src/listeners.js:423]:
```suggestion
            $('#filter').css('display', 'none').empty();
            $('.filterX').css('display', 'none');
```

File: public/src/listeners.js [hunk 2]
```diff
@@ -394,6 +399,47 @@ export default function startListeners() {
...
+        }
+        $('.filterX').css('display', 'block');
+        $('#menu').css('display', 'none');
+        let htmlIcons = '';
+        const result = circuitElementList.filter(ele => ele.toLowerCase().includes(value));
+        if(!result.length) $('#filter').text('No elements found ...');
+        else {
+            result.forEach( e => htmlIcons += createIcon(e));
+            $('#filter').html(htmlIcons);
+            $('#filter').on('click', e => selectElement($(e.target)[0].parentElement.classList[0]));
```
Comment (revathskumar) [inline public/src/listeners.js:435]:
```suggestion
            $('#filter')
              .html(htmlIcons)
              .on('click', e => selectElement($(e.target)[0].parentElement.classList[0]));
```

File: app/views/simulator/edit.html.erb [hunk 3]
```diff
@@ -70,9 +72,37 @@
     </span>
 
 
-    <div class='quick-btn'>
+    <div  ondrag="(() => {$('.quick-btn').css({'box-shadow': 'none', 'background': '#2d2d2d'})})()" class='quick-btn'>
```
Comment (revathskumar) [inline app/views/simulator/edit.html.erb:75]: Can we avoid inline js?

File: public/src/ux.js [hunk 4]
```diff
@@ -403,3 +419,55 @@ $('#octalInput').on('keyup', () => {
...
+
+    $('#moduleProperty').on('mousedown', () => {
+        $('#moduleProperty').css('z-index', '100')
+        $('.ce-panel').css('z-index', '99')
+    })
+});
+
+window.addEventListener('DOMContentLoaded', () => {
+    $('#ceMinimize').on('click', () => {
+        const markUp = `<div class='ce-hidden'>Circuit Elements<span id='ce-expand' style="right: 15px;position: absolute; cursor:pointer;"><i  onclick=" (() => {$('.modules').children().css('display', 'block'); $('.ce-hidden').remove(); $('#filter').css('display', 'none');})();" class="fas fa-external-link-square-alt"></i></span><div>`
```
[the remark this answers is not in the document]
Comment (abstrekt) [reply public/src/ux.js:445]: Exactly the event binding was not happening, that's why I used this workaround, I'll try with the solution you provided & check.

**Comment by abstrekt** [comment]
@satu0king the color theme has been integrated & going good with this PR

**Comment by abstrekt** [comment]
What could be the reason for this test to fail?
```    Assignments when user is a member should be able to make assignment project - spec.system.assignment_spec
    spec/system/assignment_spec.rb

    Failure/Error: click_on "Start Working"

    Net::ReadTimeout:
      Net::ReadTimeout with #<TCPSocket:(closed)>



    ./spec/system/assignment_spec.rb:70:in `block (3 levels) in <top (required)>'```

**Pull request closed without merging**